experimental/dds/tree2/src/feature-libraries/editable-tree-2/editableTreeTypes.ts

```diff
@@ -470,23 +464,146 @@ export interface Sequence<TTypes extends AllowedTypes> extends TreeField {
 	at(index: number): UnboxNodeUnion<TTypes>;
 
 	/**
-	 * Gets a boxed node of this field by its index.
-	 * Note that a node must exist at the given index.
+	 * Returns the index of the first occurrence of a value in the sequence, or -1 if it is not present.
+	 * @param searchElement - The element to search for.
+	 * @param fromIndex - The index at which to begin the search. If fromIndex is omitted, the search starts at index 0.
 	 */
-	boxedAt(index: number): TypedNodeUnion<TTypes>;
+	indexOf(searchElement: UnboxNodeUnion<TTypes>, fromIndex?: number): number;
 
 	/**
-	 * Calls the provided callback function on each child of this sequence, and returns an array that contains the results.
-	 * @param callbackfn - A function that accepts the child, its index, and this field.
+	 * Determines whether an sequence includes a certain element, returning true or false as appropriate.
+	 * @param searchElement - The element to search for.
+	 * @param fromIndex - The index at which to begin the search. If fromIndex is omitted, the search starts at index 0.
+	 */
+	includes(searchElement: UnboxNodeUnion<TTypes>, fromIndex?: number): boolean;
+
+	/**
+	 * Iterates the indices of the elements in the sequence.
+	 */
+	keys(): IterableIterator<number>;
+
+	/**
+	 * Iterates the values in the sequence.
+	 */
+	values(): IterableIterator<UnboxNodeUnion<TTypes>>;
+
+	/**
+	 * Iterates the elements in the sequence as tuples of index and element.
 	 */
-	map<U>(callbackfn: (value: UnboxNodeUnion<TTypes>, index: number, array: this) => U): U[];
+	entries(): IterableIterator<[number, UnboxNodeUnion<TTypes>]>;
+
+	/**
+	 * Executes a provided function once per each element in the sequence.
+	 * @param callbackFn - The function to run for each element.
+	 */
+	forEach(
+		callbackFn: (element: UnboxNodeUnion<TTypes>, index: number, sequence: this) => void,
+	): void;
+
+	/**
+	 * Returns the value of the first element in the sequence where `predicate` is true, and undefined otherwise.
+	 * @param predicate - The predicate to run on each element.
+	 */
+	find(
+		predicate: (value: UnboxNodeUnion<TTypes>, index: number, sequence: this) => boolean,
+	): UnboxNodeUnion<TTypes> | undefined;
+	find<T extends UnboxNodeUnion<TTypes>>(
+		predicate: (element: UnboxNodeUnion<TTypes>, index: number, sequence: this) => element is T,
+	): T | undefined;
+
+	/**
+	 * Returns the value of the last element in the sequence where `predicate` is true, and undefined otherwise.
+	 * @param predicate - The predicate to run on each element.
+	 */
+	findLast(
+		predicate: (value: UnboxNodeUnion<TTypes>, index: number, sequence: this) => boolean,
+	): UnboxNodeUnion<TTypes> | undefined;
+	findLast<T extends UnboxNodeUnion<TTypes>>(
+		predicate: (element: UnboxNodeUnion<TTypes>, index: number, sequence: this) => element is T,
+	): T | undefined;
+
+	/**
+	 * Returns the index of the first element in the sequence where `predicate` is true, otherwise -1.
```

Comment:
the convetion of returning -1 on not finding the result is a bad idea in a typesScript API since the type system can't help remind you to test for that. In this case returing undefined would make more sense in my opinion

Reply:
Similar argument.  In isolation, you could argue that a customer could forget to handle the -1 case and therefore having a separate sentinel value is superior.  In the context of a customer using a mix of Arrays and SharedTree lists/sequences in their code, most customers would prefer consistency.